llvm/include/llvm/Option/ArgList.h

```
@@ -280,6 +281,12 @@ class ArgList {
   /// list.
   virtual unsigned getNumInputArgStrings() const = 0;
 
+  /// getSubcommand - Return the active subcommand, if one exists.
+  LLVM_ABI StringRef getSubcommand(
+      const ArrayRef<OptTable::Command> Commands,
+      std::function<void(ArrayRef<StringRef>)> HandleMultipleSubcommands,
+      std::function<void(ArrayRef<StringRef>)> HandleOtherPositionals) const;
```

Maybe add comments above briefly explaining what these callbacks are for. Mostly requesting since the implication to me (based on the example usage) is that a user might want to just throw an error and exit. If this is the primary/only desirable case, then that might be worth noting. I personally can't think of a reason one might not want to print help and exit, but maybe someone else might have a reason to. We could probably discourage that with comments or maybe saying the callback is NoReturn if we want.